player/lua/options.lua

```diff
@@ -144,15 +144,17 @@ local function read_options(options, identifier, on_update)
             local new_opts = opt_table_copy(conf_and_default_opts)
             parse_opts(val, new_opts)
             local changelist = {}
+            local changed = false
             for key, val in pairs(new_opts) do
                 if not opt_equal(last_opts[key], val) then
                     -- copy to user
                     options[key] = opt_copy(val)
                     changelist[key] = true
+                    changed = true
                 end
             end
             last_opts = new_opts
-            if #changelist then
+            if changed then
```

Comment:
alternatively `next(changelist) ~= nil` would also detect is the dict is non-empty

Reply:
Quite right, I'm new to Lua so I didn't know about that function. Should I change the pull request?